### ISSUE TITLE
Build fails with Swift 4.2 on Ubuntu 16.04

### ISSUE THREAD

**Issue by DunnCoding** [IBM-Swift/BlueSignals#8]
We recently added a 4.2 CI build on Kitura which failed.  
It seems BlueSignals is failing to build on Swift 4.2. 

This is the error message I'm seeing: 
```swift
BlueSignals/Sources/Signals/Signals.swift:111:36: error: converting non-escaping value to 'T' may allow it to escape
                        sigAction.__sigaction_handler = unsafeBitCast(action, to: sigaction.__Unnamed_union___sigaction_handler.self)
```

**Comment by ianpartridge** [comment]
@billabt any thoughts on this - we'd like to start testing Kitura against Swift 4.2 snapshots and this blocks us.

@ddunn2 did you have a fix?

**Comment by billabt** [comment]
@ianpartridge:  I’m out of town for my daughter’s graduation until next Tuesday. All I have with me is an iPad.

**Comment by ianpartridge** [comment]
No problem, @ddunn2 will post what he found.

**Comment by DunnCoding** [comment]
Hi @billabt 

I spent some time playing around with why this is failing. It fails on both Mac and Linux it seems when building locally and this is where it fails: 

```swift
var signalAction = sigaction(__sigaction_u: unsafeBitCast(action, to: __sigaction_u.self), sa_mask: 0, sa_flags: 0)
```
The error message is:  

`error: converting non-escaping value to 'T' may allow it to escape`

I've been able to distil this down into a simple test case in a playground: 

```swift
typealias testAction = (Int32) -> Void

func testFunc(action: (Int32) -> Void) {
    unsafeBitCast(action, to: testAction.self)
}
```

On Swift 4.1 this generates a warning:  
`'unsafeBitCast' from non-escaping function type '(Int32) -> Void' to escaping function type 'testAction' (aka '(Int32) -> ()') is undefined; use 'withoutActuallyEscaping' to temporarily escape a function`

and on Swift 4.2 it just fails to compile. 

I've found a couple of solutions to this: 

This first is the warnings suggestion to use `withoutActuallyEscaping`: 

```swift
typealias testAction = (Int32) -> Void

func testFunc(action: (Int32) -> Void) {
    withoutActuallyEscaping(action) { escapableAction in
        unsafeBitCast(escapableAction, to: testAction.self)
    }
}
```

The second is to make the parameter itself `@escaping`: 
```swift
typealias testAction = (Int32) -> Void

func testFunc(action: @escaping (Int32) -> Void) {
    
    unsafeBitCast(action, to: testAction.self)
}
```

I'm not sure if either solution is viable in the scope of this project but I've also [raised a bug](https://bugs.swift.org/browse/SR-7653) for this on swift.org.
